bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -196,38 +281,116 @@ public Response update(@PathParam("thingUID") String thingUID, ThingDTO thingBea
             bridgeUID = new ThingUID(thingBean.bridgeUID);
         }
 
-        Thing thing = managedThingProvider.get(thingUIDObject);
-        if (thing == null) {
+        //
+        // ask whether the Thing exists at all, 404 otherwise
+        //
+        Thing thing = thingRegistry.get(thingUIDObject);
+        if (null == thing) {
             logger.info("Received HTTP PUT request for update at '{}' for the unknown thing '{}'.", uriInfo.getPath(),
                     thingUID);
-            return Response.status(Status.NOT_FOUND).build();
+            return getThingNotFoundResponse(thingUID);
         }
 
-        thing.setBridgeUID(bridgeUID);
+        //
+        // ask whether the Thing exists as a managed thing, so it can get
+        // updated, 409 otherwise
+        //
+        Thing managed = managedThingProvider.get(thingUIDObject);
+        if (null == managed) {
+            logger.info("Received HTTP PUT request for update at '{}' for an unmanaged thing '{}'.", uriInfo.getPath(),
+                    thingUID);
+            return getThingResponse(Status.CONFLICT, thing,
+                    "Cannot update Thing " + thingUID + ". Maybe it is not managed.");
+        }
 
+        //
+        // only process if Thing is known to be managed, so it can get updated
+        //
+        thing.setBridgeUID(bridgeUID);
         updateConfiguration(thing, getConfiguration(thingBean));
 
-        managedThingProvider.update(thing);
+        // update, returns null in case of failure
+        Thing oldthing = managedThingProvider.update(thing);
 
-        return Response.ok().build();
+        //
+        // could not update?
+        //
+        if (null == oldthing) {
+            // must be an unknown error, dunno
+            return getThingResponse(Status.INTERNAL_SERVER_ERROR, thing,
```

Comment:
update only returns null, if no thing with the given UID exists. So it should be a client error (4xx).